csi/src/node.rs

```diff
@@ -161,443 +179,405 @@ impl NodeImpl {
 
-        // Check arguments
-        let vol_id = req.get_volume_id();
-        if vol_id.is_empty() {
-            return util::fail(
-                &ctx,
-                sink,
-                RpcStatusCode::INVALID_ARGUMENT,
-                &anyhow!("volume ID missing in request"),
-            );
-        }
-        let target_path = req.get_target_path();
-        if target_path.is_empty() {
-            return util::fail(
-                &ctx,
-                sink,
-                RpcStatusCode::INVALID_ARGUMENT,
-                &anyhow!("target path missing in request"),
-            );
-        }
-
-        let volume = match self.meta_data.get_volume_by_id(vol_id) {
-            Some(v) => v,
-            None => {
-                return util::fail(
-                    &ctx,
-                    sink,
-                    RpcStatusCode::NOT_FOUND,
-                    &anyhow!(format!("failed to find volume ID={}", vol_id)),
-                )
+        let task = async move {
+            // Check arguments
+            let vol_id = req.get_volume_id();
+            if vol_id.is_empty() {
+                return Err((
+                    RpcStatusCode::INVALID_ARGUMENT,
+                    anyhow!("volume ID missing in request"),
+                ));
+            }
+            let target_path = req.get_target_path();
+            if target_path.is_empty() {
+                return Err((
+                    RpcStatusCode::INVALID_ARGUMENT,
+                    anyhow!("target path missing in request"),
+                ));
             }
-        };
 
-        let r = NodeUnpublishVolumeResponse::new();
-        // Do not return error for non-existent path, repeated calls OK for idempotency
-        // if unistd::geteuid().is_root() {
-        let delete_res = self
-            .meta_data
-            .delete_volume_one_bind_mount_path(vol_id, target_path);
-        let mut pre_mount_path_set = match delete_res {
-            Ok(s) => s,
-            Err(e) => {
-                warn!(
-                    "failed to delete mount path={} of volume ID={} from etcd, \
+            let volume = self_inner.meta_data.get_volume_by_id(vol_id).await.ok_or((
+                RpcStatusCode::NOT_FOUND,
+                anyhow!(format!("failed to find volume ID={}", vol_id)),
+            ))?;
+
+            let r = NodeUnpublishVolumeResponse::new();
+            // Do not return error for non-existent path, repeated calls OK for idempotency
+            // if unistd::geteuid().is_root() {
+            let delete_res = self_inner
+                .meta_data
+                .delete_volume_one_bind_mount_path(vol_id, target_path)
+                .await;
+            let mut pre_mount_path_set = match delete_res {
+                Ok(s) => s,
+                Err(e) => {
+                    warn!(
+                        "failed to delete mount path={} of volume ID={} from etcd, \
                         the error is: {}",
```

Comment:
extra 4 spaces indentation before the second text line

Reply:
Done